base/src/net.ext.c

```diff
@@ -276,6 +276,12 @@ void on_connect6(uv_connect_t *connect_req, int status) {
         char errmsg[1024] = "Failed to write to TCP socket: ";
         uv_strerror_r(r, errmsg + strlen(errmsg), sizeof(errmsg)-strlen(errmsg));
         log_warn(errmsg);
+        if (strstr(errmsg, "bad file descriptor")) {
```

Comment:
`r` contains a `UV_something` error, better inspect it than doing string comparison (le expensive). See uv.h